File: q2169/q2169.py
class Solution:
    def countOperations(self, num1: int, num2: int) -> int:
        ops = 0
        while num1 and num2:
            # Better performance than simulating the process
            ops += num1 // num2
            num1 %= num2
            num1, num2 = num2, num1
        return ops

File: q2169/q2169.java
class Solution {
    public int countOperations(int num1, int num2) {
        int ops = 0; 
        while (num1 != 0 && num2 != 0) {
            ops += num1 / num2;
            num1 %= num2;
            int temp = num2;
            num2 = num1; 
            num1 = temp;
        }
        return ops;
    }
}
